decompiler/pipeline/controlflowanalysis/restructuring_commons/condition_aware_refinement_commons/switch_extractor.py

```diff
@@ -64,7 +56,20 @@ def _extract_switch_node_from_branch(self, branch: Union[TrueNode, FalseNode]) -
         :param branch: The branch from which we extract the switch node.
         :return: If we introduce a new sequence node, then return this node, otherwise return None.
         """
-        if len(self._current_cond_node.children) != 2:
-            self.asforest.replace_condition_node_by_single_branch(self._current_cond_node)
+        assert isinstance(condition_node := branch.parent, ConditionNode), "The parent of a true/false-branch must be a condition node!"
+        if len(condition_node.children) != 2:
+            self.asforest.replace_condition_node_by_single_branch(condition_node)
         else:
-            self.asforest.extract_branch_from_condition_node(self._current_cond_node, branch, False)
+            self.asforest.extract_branch_from_condition_node(condition_node, branch, False)
+
+    def _extract_switch_from_first_or_last_child_of(self, sequence_node: SeqNode, condition: LogicCondition) -> bool:
```

Comment:
btw. and regarding my above comment: Here you are also returning True/False in case of succeeded or failed extraction without the function being names as above

Reply:
renamed